index.d.ts

```diff
@@ -0,0 +1,26 @@
+declare module 'waveform-data' {
+
+    type WaveformDataChannelType = {
+        min_sample: (index: number) => number
+        max_sample: (index: number) => number
+        min_array: (index: number) => Array<number>
+        max_array: (index: number) => Array<number>
+    }
+
+    type WaveformDataType = {
+        resample: (options: number | {width: number, scale: number}) => WaveformDataType
+        concat: (args?: Array<WaveformDataType>) => WaveformDataType
+        length: number
+        bits: number
+        duration: number
+        pixels_per_second: number
+        seconds_per_pixel: number
+        channels: number
+        channel: (index: number) => WaveformDataChannelType
+        sample_rate: number
+        scale: number
+        at_time: (time: number) => number
+        time: (time: number) => number
+    }
+    static const create: (data: any) => WaveformDataType
```

Comment:
`data` can only be `ArrayBuffer` or an object with fields as described [here](https://github.com/bbc/audiowaveform/blob/master/doc/DataFormat.md#json-data-format-json). We could add a type declaration to valdate the object structure.